pkg/cmd/issue/list/list.go

```diff
@@ -174,6 +174,10 @@ func listRun(opts *ListOptions) error {
 		return err
 	}
 
+	if len(listResult.Issues) == 0 {
+		return prShared.ListNoResults(opts.IO, ghrepo.FullName(baseRepo), "issue", !filterOptions.IsDefault())
+	}
```

Comment:
While this pertains to listing empty result sets I would like to see this change and the one to `pr list` extracted into a separate PR as it doesn't use `HandleNoResults` and I would like to keep this one more focussed since it has grown in size.

Reply:
The function I added `prShared.ListNoResults` _does_ use `HandleNoResults` (now renamed to `ErrorNoResults`) in its implementation. I just kept them in a shared component to keep the logic for which error message to display factored out.

Happy to revert those changes if you'd like?

Reply:
That's a good point. I would say that `shared.ListHeader` should remove the no results found logic then since we now have a dedicated function for that.